yapio.c

```diff
@@ -2135,6 +2180,37 @@ yapio_display_result(const yapio_test_ctx_t *ytc)
     }
 }
 
+/* dump yapioMyRank's metadata array into a file */
+static void
+yapio_store_md_final_state()
+{
+    /* each rank writes its own md file '{filename}-md-{rank}' */
+    char md_file[PATH_MAX + 10];
+    snprintf(md_file, PATH_MAX, "%s-md-%d", yapioTestFileName, yapioMyRank);
+
+    yapio_test_group_t *ytg = yapioMyTestGroup;
+    const size_t nblks_per_rank = ytg->ytg_num_blks_per_rank;
+
+    /* use metadata from last iteration */
+    int test_ctx_idf = ytg->ytg_num_contexts - 1;
+    yapio_test_ctx_t *ytc = &ytg->ytg_contexts[test_ctx_idf];
+
+    FILE *file = fopen(md_file, "w+");
```

Comment:
"w+" is really necessary ? just "w" should be fine

Reply:
opening for writing only so "w" should be enough